dot/network/service.go

```diff
@@ -149,13 +149,11 @@ func NewService(cfg *Config) (*Service, error) {
 	// initially allocate as many buffers as liekly necessary which is the number inbound streams we will have,
 	// which should equal average number of peers times the number of notifications protocols, which is currently 3.
 	var bufPool *sizedBufferPool
-	if cfg.noPreAllocate {
-		bufPool = &sizedBufferPool{
-			c: make(chan *[maxMessageSize]byte, cfg.MinPeers*3),
-		}
-	} else {
-		bufPool = newSizedBufferPool(cfg.MinPeers*3, cfg.MaxPeers*3)
+	preAllocatedInPool := 0
+	if !cfg.noPreAllocate {
+		preAllocatedInPool = cfg.MinPeers * 3
 	}
+	bufPool = newSizedBufferPool(preAllocatedInPool, cfg.MinPeers*3)
```

Comment:
the maximum should be `cfg.MaxPeers*3`

Reply:
Ohhh thank you, blind mee!! I'll fix that.

Reply:
Wait that's confusing too, why would the channel capacity `make(chan *[maxMessageSize]byte, cfg.MinPeers*3),` above not be `cfg.MaxPeers*3` as well then? Pre-allocation or not, I don't see how this should influence the channel capacity? 🤔

Reply:
you mean in the `!cfg.noPreAllocate` case? that's only used for testing, so there isn't really a point in pre-allocating `cfg.MaxPeers*3` buffers

Reply:
No, I mean why is the maximum channel capacity (aka `max`, different from the pre-allocated buffers in the channel queue) different.

When noPreAllocated (production) is true, the capacity is `cfg.MinPeers*3`. When it's false (for testing), it's `cfg.MaxPeers*3` 🤔  That looks strange right? Didn't we want to pre-allocate `cfg.MinPeers*3` and have a max of `cfg.MinPeers*3` for testing as well?

Reply:
`noPreAllocated` is always false in production, as we want to pre-allocate, it's only set to true in the tests. in testing, it's fine to have a maximum channel capacity of `cfg.MinPeers*3` as we won't really be using many buffers (we could probably set it to 3 or something even). when it's false, in production, we want to be able to store `cfg.MaxPeers*3` buffers potentially as that is the maximum amount of buffers we woudl ever need (# of peers * 3 streams per peer)

Reply:
@qdm12 can you update this when you get a chance? everything else looks good :)

Reply:
Done :+1: Yeah I had it all wrong, it should be fixed now.